test/test_gpu3.cpp

```diff
@@ -9154,6 +9154,56 @@ TEST_F(NVFuserTest, FusionRecomputePersistentBuffer_CUDA) {
   testValidate(fusion, cg_outputs, inputs, outputs, __LINE__, __FILE__);
 }
 
+// Based on FusionTestWarnRegisterSpill_CUDA but modified to test OptionsGuard
+TEST_F(NVFuserTest, FusionOptionsGuard_CUDA) {
+  const int hidden_size = 1024 * 10;
+  std::unique_ptr<Fusion> fusion_ptr = std::make_unique<Fusion>();
+  Fusion& fusion = *fusion_ptr.get();
+  FusionGuard fg(&fusion);
+  const float kEps = 1e-5;
+  Double* eps_ptr = IrBuilder::create<Double>(kEps);
+  std::vector<int64_t> input_shape{2048, hidden_size};
+  std::vector<int64_t> norm_shape{hidden_size};
+
+  auto input = makeSymbolicTensor(input_shape.size());
+  fusion.addInput(input);
+  auto result = layer_norm(input, norm_shape, nullptr, nullptr, eps_ptr);
+  fusion.addOutput(result.output);
+  fusion.addOutput(result.mean);
+  fusion.addOutput(result.invstd);
+
+  auto options = at::TensorOptions().dtype(at::kFloat).device(at::kCUDA, 0);
+  at::Tensor aten_input = at::randn(input_shape, options);
+  c10::optional<at::Tensor> aten_weight = c10::nullopt;
+  c10::optional<at::Tensor> aten_bias = c10::nullopt;
+  auto aten_outputs = at::native_layer_norm(
+      aten_input, norm_shape, aten_weight, aten_bias, kEps);
+
+  // generate persistent kernel
+  auto persistent_params = getPersistentHeuristics(&fusion, {aten_input});
+  TORCH_CHECK(persistent_params, "Persistent schedule was not generated!");
```

Comment:
nit: should we prefer `EXPECT_TRUE` over `TORCH_CHECK`? This is just a matter of coding style, nothing about behavior. BTW, you can leave failure message by using `EXPECT_TRUE(...) << ...;`

Reply:
Looks like there's also `ASSERT_TRUE`, which immediately stops the execution of the test. `ASSERT_TRUE` seems more appropriate as it usually doesn't make any sense to continue the execution of the test.